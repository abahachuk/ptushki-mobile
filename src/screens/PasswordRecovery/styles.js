import { StyleSheet } from "react-native";

/* eslint  import/prefer-default-export: 0 */
export const styles = StyleSheet.create({
  container: {
    width: "100%",
    flex: 1,
    justifyContent: "space-between",
    alignItems: "center",
    paddingRight: 20,
    paddingLeft: 20
  },
  main: {
    width: "100%"
  },
  logoImg: {
    width: 300,
    height: 280,
    alignSelf: "center"
  },
  headerText: {
    fontSize: 34,
    textAlign: "left"
  },
  hintText: {
    paddingTop: 10,
    paddingBottom: 15
  },
  resetPswBtn: {
    width: "100%",
    marginTop: 15
  },
  footer: {
    width: "100%"
  },
  backButton: {
    marginBottom: 10
  }
});