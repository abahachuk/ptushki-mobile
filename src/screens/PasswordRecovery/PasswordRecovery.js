import React, { useState } from "react";
import PropTypes from "prop-types";
import { Text, KeyboardAvoidingView, ScrollView } from "react-native";

import { Button, Input } from "../../components";

import { styles } from "./styles";

const fields = {
  email: "email"
};

const PasswordRecovery = props => {
  const { emailDefault, hintText } = props;

  const [{ email }] = useState({
    [fields.email]: emailDefault
  });

  const onRecoveryPress = () => {
    props.onSubmit();
    props.navigation.navigate("passwordResetDone", {
      origin: "passwordRecovery",
      statusText: "Мы выслали вам новый пароль!",
      hintText:
        "Проверьте свой E-mail, чтобы получить новый пароль и введите его при входе. После этого у вас будет возможность его заменить"
    });
  };
  const onBackPress = () => {
    props.navigation.goBack();
  };

  return (
    <ScrollView
      contentContainerStyle={styles.scrollContainer}
      showsVerticalScrollIndicator={false}
      enabled
    >
      <KeyboardAvoidingView style={styles.container} enabled>
        <Text style={styles.headerText}>Восстановление пароля</Text>
        <Text style={styles.hintText}>{hintText}</Text>
        <Input value={email} label="E-mail" textContentType="emailAddress" />

        <Button
          caption="Выслать новый пароль"
          onPress={onRecoveryPress}
          appearance="Dark"
        />
        <Button caption="НАЗАД" onPress={onBackPress} appearance="Borderless" />
      </KeyboardAvoidingView>
    </ScrollView>
  );
};

PasswordRecovery.navigationOptions = {
  title: "Восстановление пароля",
  header: null
};

PasswordRecovery.propTypes = {
  emailDefault: PropTypes.string,
  hintText: PropTypes.string,
  onSubmit: PropTypes.func,
  navigation: PropTypes.shape({
    goBack: PropTypes.func,
    navigate: PropTypes.func
  }).isRequired
};
PasswordRecovery.defaultProps = {
  emailDefault: "",
  hintText: "Введите ваш E-mail, чтобы мы выслали новый пароль",
  onSubmit: () => {}
};

export default PasswordRecovery;
