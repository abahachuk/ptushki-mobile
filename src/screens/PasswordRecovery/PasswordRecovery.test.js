import React from 'react';
import renderer from 'react-test-renderer';

import PasswordRecovery from './PasswordRecovery';
/* eslint  no-undef: 0 */
describe('Login screen', () => {
  test('renders correctly', () => {
    const navigation = { navigate: jest.fn() };
    const tree = renderer.create(<PasswordRecovery navigation={navigation} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
