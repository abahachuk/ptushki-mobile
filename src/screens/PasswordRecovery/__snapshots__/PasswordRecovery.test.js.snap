// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen renders correctly 1`] = `
<RCTScrollView
  enabled={true}
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "space-between",
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        [missing "en.passwordRecovery.recovery" translation]
      </Text>
      <Text>
        [missing "en.passwordRecovery.instructionText" translation]
      </Text>
      <View
        style={Array []}
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.passwordRecovery.email" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="emailAddress"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={Array []}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 6,
              "paddingBottom": 13,
              "paddingTop": 13,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#546e7a",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#546e7a",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          [missing "en.passwordRecovery.sendNewPassword" translation]
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 6,
              "paddingBottom": 13,
              "paddingTop": 13,
              "width": "100%",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#546e7a",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              undefined,
            ]
          }
        >
          [missing "en.passwordRecovery.back" translation]
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
